openupgrade_scripts/scripts/hr/16.0.1.1/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,75 @@
+---Models in module 'hr'---
+model hr.contract.type (moved from hr_contract)
+# DONE: pre-migration: moved from hr_contract to hr
+
+---Fields in module 'hr'---
+hr           / hr.contract.type         / __last_update (datetime)      : previously in module hr_contract
+hr           / hr.contract.type         / _order                        : previously in module hr_contract
+hr           / hr.contract.type         / display_name (char)           : previously in module hr_contract
+hr           / hr.contract.type         / name (char)                   : previously in module hr_contract
+# DONE: pre-migration: moved from hr_contract to hr
+
+hr           / hr.contract.type         / sequence (integer)            : NEW
+# NOTHING TO DO
+
+hr           / hr.department            / master_department_id (many2one): NEW relation: hr.department, isfunction: function, stored
+hr           / hr.department            / parent_path (char)            : NEW
+hr           / hr.department            / plan_ids (one2many)           : NEW relation: hr.plan
+# NOTHING TO DO
+
+hr           / hr.employee              / mobile_phone (char)           : now a function
+hr           / hr.employee              / work_contact_id (many2one)    : NEW relation: res.partner
+hr           / hr.employee              / work_email (char)             : now a function
+# TODO: post-migration: create work_contact_id for each employee
+
+hr           / hr.job                   / active (boolean)              : NEW hasdefault: default
+hr           / hr.job                   / contract_type_id (many2one)   : NEW relation: hr.contract.type
+hr           / hr.job                   / state (selection)             : DEL required, selection_keys: ['open', 'recruit']
+# NOTHING TO DO: hr.job does not work with state anymore.
+
+hr           / hr.plan                  / company_id (many2one)         : NEW relation: res.company, hasdefault: default
+hr           / hr.plan                  / department_id (many2one)      : NEW relation: hr.department
+# NOTHING TO DO
+
+hr           / hr.plan                  / plan_activity_type_ids (many2many): table is now 'False' ('hr_plan_hr_plan_activity_type_rel')
+hr           / hr.plan                  / plan_activity_type_ids (many2many): type is now 'one2many' ('many2many')
+# DONE: pre-migration and post-migration: move data from many2many table to plan_id colomn in hr.plan.activity.type
+
+hr           / hr.plan.activity.type    / company_id (many2one)         : NEW relation: res.company, hasdefault: default
+# NOTHING TO DO: not a required field so no default to set
+
+hr           / hr.plan.activity.type    / plan_id (many2one)            : NEW relation: hr.plan
+# DONE: see plan_activity_type_ids from hr.plan
```

Comment:
You can reorder this line to put it joined with the one2many field.

Reply:
Done.